File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/admin/refresh/RefreshResponse.java
package fr.myprysm.vertx.elasticsearch.action.admin.refresh;

import fr.myprysm.vertx.elasticsearch.action.support.ShardFailure;
import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.elasticsearch.rest.RestStatus;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@DataObject(generateConverter = true)
public class RefreshResponse {

    private int totalShards;
    private int successfulShards;
    private int failedShards;
    private List<ShardFailure> shardFailures;
    private RestStatus status;

    public RefreshResponse(RefreshResponse other) {
        totalShards = other.totalShards;
        successfulShards = other.successfulShards;
        failedShards = other.failedShards;
        shardFailures = other.shardFailures;
        status = other.status;
    }

    /**
     * Build a new <code>RefreshResponse</code> from a <code>JsonObject</code>.
     *
     * @param json the <code>JsonObject</code>
     */
    public RefreshResponse(JsonObject json) {
        RefreshResponseConverter.fromJson(json, this);
    }

    /**
     * Transforms the <code>RefreshResponse</code> into a <code>JsonObject</code>.
     *
     * @return the <code>JsonObject</code>
     */
    public JsonObject toJson() {
        JsonObject json = new JsonObject();
        RefreshResponseConverter.toJson(this, json);
        return json;
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/admin/refresh/RefreshResponse.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.admin.refresh

import fr.myprysm.vertx.elasticsearch.action.admin.refresh.RefreshResponse
import fr.myprysm.vertx.elasticsearch.action.support.ShardFailure
import org.elasticsearch.rest.RestStatus

fun RefreshResponse(
  failedShards: Int? = null,
  shardFailures: Iterable<fr.myprysm.vertx.elasticsearch.action.support.ShardFailure>? = null,
  status: RestStatus? = null,
  successfulShards: Int? = null,
  totalShards: Int? = null): RefreshResponse = fr.myprysm.vertx.elasticsearch.action.admin.refresh.RefreshResponse().apply {

  if (failedShards != null) {
    this.setFailedShards(failedShards)
  }
  if (shardFailures != null) {
    this.setShardFailures(shardFailures.toList())
  }
  if (status != null) {
    this.setStatus(status)
  }
  if (successfulShards != null) {
    this.setSuccessfulShards(successfulShards)
  }
  if (totalShards != null) {
    this.setTotalShards(totalShards)
  }
}

